extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -0,0 +1,361 @@
+						data.push([difference.name, difference.sourceValue, this.SchemaCompareActionMap[difference.updateAction], difference.targetValue]);
+					}
+				}
+			});
+		}
+
+		return data;
+	}
+
+	private getAggregatedScript(diffEntry: azdata.DiffEntry, getSourceScript: boolean): string {
+		let script = '';
+		if (diffEntry !== null) {
+			script += getSourceScript ? diffEntry.sourceScript : diffEntry.targetScript;
+			diffEntry.children.forEach(child => {
+				let childScript = this.getAggregatedScript(child, getSourceScript);
+				if (childScript !== 'null') {
+					script += childScript;
+				}
+			});
+		}
+		return script;
+	}
+
+	private reExecute(): void {
+		this.flexModel.removeItem(this.splitView);
+		this.flexModel.removeItem(this.noDifferencesLabel);
+		this.flexModel.addItem(this.loader, { CSSStyles: { 'margin-top': '30px' } });
+		this.diffEditor.updateProperties({
+			contentLeft: '\n',
+			contentRight: '\n'
+		});
+		this.differencesTable.selectedRows = null;
+		this.resetButtons();
+		this.execute();
+	}
+
+	private createCompareButton(view: azdata.ModelView): void {
+		let runIcon = path.join(__dirname, '.', 'media', 'compare.svg');
+
+		this.compareButton = view.modelBuilder.button().withProperties({
+			label: localize('schemaCompare.compareButton', 'Compare'),
+			iconPath: runIcon,
+			title: localize('schemaCompare.compareButtonTitle', 'Compare')
+		}).component();
+
+		this.compareButton.onDidClick(async (click) => {
+			this.reExecute();
+		});
+	}
+
+	private createGenerateScriptButton(view: azdata.ModelView): void {
+		let fileIcon = path.join(__dirname, '.', 'media', 'generate-script.svg');
+
+		this.generateScriptButton = view.modelBuilder.button().withProperties({
+			label: localize('schemaCompare.generateScriptButton', 'Generate script'),
+			iconPath: fileIcon
+		}).component();
+
+		this.generateScriptButton.onDidClick(async (click) => {
+			// get file path
+			let now = new Date();
+			let datetime = now.getFullYear() + '-' + (now.getMonth() + 1) + '-' + now.getDate() + '-' + now.getHours() + '-' + now.getMinutes() + '-' + now.getSeconds();
+			let defaultFilePath = path.join(os.homedir(), this.targetName + '_Update_' + datetime + '.sql');
+			let fileUri = await vscode.window.showSaveDialog(
+				{
+					defaultUri: vscode.Uri.file(defaultFilePath),
+					saveLabel: localize('schemaCompare.saveFile', 'Save'),
+					filters: {
+						'SQL Files': ['sql'],
+					}
+				}
+			);
+
+			if (!fileUri) {
+				return;
+			}
+
+			let service = await SchemaCompareResult.getService('MSSQL');
+			let result = await service.schemaCompareGenerateScript(this.comparisonResult.operationId, this.targetEndpointInfo.databaseName, fileUri.fsPath, azdata.TaskExecutionMode.execute);
+			if (!result || !result.success) {
+				vscode.window.showErrorMessage(
+					localize('schemaCompare.generateScriptErrorMessage', "Generate script failed '{0}'", result.errorMessage ? result.errorMessage : 'Unknown'));
```

Comment:
result might be undefined here, result.error.errorMessage is used without checking result object

Reply:
done